numba/cpython/unicode.py

```diff
@@ -2361,16 +2362,23 @@ def impl(n):
                 flag = True
             if n == 0:
                 return '0'
-            l = []
+            length = flag + 1 + int(np.floor(np.log10(n)))
+            kind = PY_UNICODE_1BYTE_KIND
+            char_width = _kind_to_byte_width(kind)
+            s = _malloc_string(kind, char_width, length, True)
+            const = ord('0')
+            if flag:
+                _set_code_point(s, 0, ord('-'))
+            idx = length - 1
             while n > 0:
-                c = chr(ord('0') + (n % ten))
+                c = const + (n % ten)
                 n = n // ten
```

Comment:
Probably doesn't make any difference, but fewer operators as:
```suggestion
                n, digit = divmod(n, ten)
                c = const + digit
```

Reply:
heh, yes, thanks, also, pretty sure Numba internally will just use divmod for doing `%` and `//` and LLVM will sort this out. Will make this change though as it's neater.